src/python/pants/engine/process.py

```diff
@@ -113,7 +124,18 @@ def __init__(
         self.jdk_home = jdk_home
         self.is_nailgunnable = is_nailgunnable
         self.execution_slot_variable = execution_slot_variable
-        self.cache_failures = cache_failures
+
+        deprecated_conditional(
+            predicate=lambda: cache_failures is not None,
+            removal_version="2.4.0.dev1",
+            entity_description="Process.cache_failures",
+            hint_message="Use `Process.cache_scope` instead.",
+        )
+        if cache_failures is not None:
+            cache_scope = (
```

Comment:
Should an error be raised here if cache_scope was passed in and doesn't match your calculation? If not - as things are - you just silently stomped the requested cache_scope value. Given the deprecation it seems like the right answer is to fail fast if both values are passed in - its one or the other, and preferably the non-deprecated `cache_scope`.

Reply:
I figured that because there is no way to use `cache_failures` without triggering the deprecation, that passing both would not be a situation you'd stay in long anyway (since that would trigger the deprecation).